src/pymanopt/manifolds/euclidean.py

```diff
@@ -107,11 +107,6 @@ def __init__(
         dimension = math.prod(shape)
         super().__init__(name, dimension, *shape, backend=backend)
 
-        @RiemannianSubmanifold.backend.setter
-        def _(self, backend: Backend):
-            assert backend.is_dtype_real()
-            super().backend = backend
-
 
 class ComplexEuclidean(_Euclidean):
```

Comment:
With what you propose, maybe it's time to get rid of these `ComplexXXXX` manifolds?

Reply:
Here my thoughts on the matter:
- The manifold dimension still changes between the real and complex case. I think we could work around this by making the dimension an `@property`.
- Currently, each manifold class has an `IS_COMPLEX` attribute indicating its realness. It is used in the `set_compatible_backend()` to choose what version of the backend (real or complex) to use. Note there are manifolds that can have both real and complex dtypes (e.g. `Euclidean`), but there also manifolds that can only be one (e.g. `SymmetricPositiveDefinite` can only be real and ` SpecialHermitianPositiveDefinite` can only be complex). We should also be able to deal with this but it would require replacing `IS_COMPLEX` with (for example) an attribute listing the possible realness of the manifold.
- Nevertheless, removing these classes directly would be a huge breaking change, and the effort required to mitigate it (e.g. by declaring a bunch of deprecated wrappers) would be considerable.

Overall, this would be a nice product of the work around the numerical backends. However, it is still independent of our current objectives and I think we can postpone until after the [`develop_backends`](https://github.com/pymanopt/pymanopt/tree/develop_backends) branch is merged.